packages/assert/src/assert.js

```diff
@@ -0,0 +1,378 @@
+  }
+  detailsNote.annotate(error, ErrorInfo.NOTE);
+};
+freeze(fail);
+
+// Don't freeze or export `baseAssert` until we add methods.
+// TODO If I change this from a `function` function to an arrow
+// function, I seem to get type errors from TypeScript. Why?
+/**
+ * The `assert` function itself.
+ *
+ * @param {*} flag The truthy/falsy value
+ * @param {Details} [optDetails] The details to throw
+ * @returns {asserts flag}
+ */
+function baseAssert(flag, optDetails = details`Check failed`) {
+  if (!flag) {
+    throw fail(optDetails);
+  }
+}
+
+/**
+ * The `assert.equal` method
+ *
+ * Assert that two values must be `Object.is`.
+ * @param {*} actual The value we received
+ * @param {*} expected What we wanted
+ * @param {Details} [optDetails] The details to throw
+ * @returns {void}
+ */
+const equal = (
+  actual,
+  expected,
+  optDetails = details`Expected ${actual} is same as ${expected}`,
+) => {
+  baseAssert(isSame(actual, expected), optDetails);
+};
+freeze(equal);
+
+/**
+ * The `assert.typeof` method.
+ *
+ * Assert an expected typeof result.
+ * @type {AssertTypeof}
+ * @param {any} specimen The value to get the typeof
+ * @param {string} typename The expected name
+ * @param {Details} [optDetails] The details to throw
+ */
+const assertTypeof = (specimen, typename, optDetails) => {
+  baseAssert(
+    typeof typename === 'string',
+    details`${q(typename)} must be a string`,
+  );
+  if (optDetails === undefined) {
+    // Like
+    // ```js
+    // optDetails = details`${specimen} must be ${q(an(typename))}`;
+    // ```
+    // except it puts the typename into the literal part of the template
+    // so it doesn't get quoted.
+    optDetails = details(['', ` must be ${an(typename)}`], specimen);
+  }
+  equal(typeof specimen, typename, optDetails);
+};
+freeze(assertTypeof);
+
+// Note that "assert === baseAssert"
+/**
+ * assert that expr is truthy, with an optional details to describe
+ * the assertion. It is a tagged template literal like
+ * ```js
+ * assert(expr, details`....`);`
+ * ```
+ * If expr is falsy, then the template contents are reported to
+ * the console and also in a thrown error.
+ *
+ * The literal portions of the template are assumed non-sensitive, as
+ * are the `typeof` types of the substitution values. These are
+ * assembled into the thrown error message. The actual contents of the
+ * substitution values are assumed sensitive, to be revealed to
+ * the console only. We assume only the virtual platform's owner can read
+ * what is written to the console, where the owner is in a privileged
+ * position over computation running on that platform.
+ *
+ * The optional `optDetails` can be a string for backwards compatibility
+ * with the nodejs assertion library.
+ * @type {typeof baseAssert & { typeof: AssertTypeof, fail: typeof fail, equal: typeof equal, note: typeof assertNote }}
+ */
+const assert = assign(baseAssert, {
+  equal,
+  fail,
+  typeof: assertTypeof,
+  note: assertNote,
```

Comment:
Looks like we named `assertTypeof` to avoid a keyword identifier collision. Perhaps, like `equal` and `fail`, we don’t need this for `note`.